src/fcp/codegen.py

```diff
@@ -7,52 +7,56 @@
 import pkgutil
 import sys
 
-from .result import Ok, Error, result_shortcut
+from typing import Any, Union
+from types import ModuleType
+
+
+from .result import Ok, Error, Result, result_shortcut
 
 
 class CodeGenerator:
     """Base class for generators."""
 
-    def __init__(self):
+    def __init__(self) -> None:
         pass
 
-    def gen(self, fcp, templates, skels, output_path):
+    def gen(self, fcp: Result, templates: Any, skels: Any, output_path: str) -> None:
         """Function called from fcp to trigger generator. Do not override."""
 
         self.verify(fcp).unwrap()
 
         self.output_path = pathlib.Path(output_path)
 
-        for path, content in self.generate(fcp, templates, skels).items():
+        for path, content in self._generate(fcp, output_path, templates, skels).items():
             logging.info(f"Generating {path}")
             os.makedirs(path.parent, exist_ok=True)
             with open(path, "w", encoding="utf-8") as file:
                 file.write(content)
 
-    def verify(self, fcp):
+    def verify(self, fcp: Any) -> Union[Ok, Error]:
         if fcp is not None:
             return Ok(())
         else:
             return Error(["Received a None object instead of FcpV2"])
 
-    def generate(self, fcp, output_path, templates, skel):
+    def _generate(self, fcp: Any, output_path: str, templates: Any, skel: Any) -> Any:
```

Comment:
why the name change?

Reply:
This may not be the best design, but, the generate method is supposed to be overridden by the plugins. So it really should be the public method. It is `gen` that is the private method, only called from inside `CodeGenerator`